src/app/view/directives/dot-maxlength/dot-maxlength.directive.ts

```diff
@@ -7,29 +9,78 @@ import { Directive, ElementRef, HostListener, Input } from '@angular/core';
 @Directive({
     selector: '[dotMaxlength]'
 })
-export class DotMaxlengthDirective {
+export class DotMaxlengthDirective implements OnInit, OnDestroy {
     private _maxLength: number;
+    private events = ['paste', 'keypress'];
+    private destroy$: Subject<boolean> = new Subject<boolean>();
 
     constructor(private el: ElementRef) {}
 
+    ngOnInit() {
+        const eventStreams = this.events.map(ev => fromEvent(this.el.nativeElement, ev));
+        const allEvents$ = merge(...eventStreams);
+        allEvents$
+            .pipe(
+                takeUntil(this.destroy$),
+                tap((keyboardEvent: Event) => {
+                    if (!this.isValidAction(keyboardEvent)) {
+                        keyboardEvent.preventDefault();
+                    }
+                }),
+                delay(1),
+                filter(() => this.el.nativeElement.textContent.length > this._maxLength),
+                tap(() => this.reduceText())
+            )
+            .subscribe();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
     @Input()
     set dotMaxlength(maxLength: number) {
         this._maxLength = maxLength || 255;
     }
 
+    private isValidAction(keyboardEvent: Event): boolean {
+        return (
+            this.el.nativeElement.textContent.length < this._maxLength ||
+            this.isAllowedKeyCode(keyboardEvent) ||
```

Comment:
why are we checking this?

Reply:
because in FF, when you do any of these will fire an event, and since we are preventing default if the string is with full length, those event get blocked and we don't want that.